specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2023-12-10-preview/cloudhsm.json

```diff
@@ -0,0 +1,967 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Cloud HSM Resource Provider",
+    "description": "The Azure management API provides a RESTful set of web services that interact with Azure Cloud HSM RP.",
+    "version": "2023-12-10-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}": {
+      "put": {
+        "tags": [
+          ""
+        ],
+        "operationId": "CloudHsmClusters_CreateOrUpdate",
+        "description": "Create or Update a Cloud HSM Cluster in the specified subscription.",
```

Comment:
Put is only for Create, Patch is for Update.

Reply:
Both PUT and PATCH can be used to update resources. ARM templates always use PUT, including for updates to existing resources.

Reply:
There is a validation running on the description, removing the update will cause the validation to fail.